### PR TITLE
COMMERCE-2401 Fixed default account roles association

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

File: commerce-account-api/src/main/java/com/liferay/commerce/account/configuration/CommerceAccountServiceConfiguration.java [hunk 1]
```diff
@@ -37,4 +37,7 @@
 	)
 	public boolean applyDefaultRoleToExistingUsers();
 
+	@Meta.AD(deflt = "", name = "site-roles", required = false)
```
Comment (ethanbustad) [inline commerce-account-api/src/main/java/com/liferay/commerce/account/configuration/CommerceAccountServiceConfiguration.java]: I believe `deflt = ""` is the default, so there's no need to specify this.

File: commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java [hunk 2]
```diff
@@ -143,26 +147,45 @@ public void addCommerceAccountUserRels(
...
+		String[] siteRoles = commerceAccountServiceConfiguration.siteRoles();
+
+		if ((siteRoles == null) && ArrayUtil.isEmpty(siteRoles)) {
+			return;
 		}
 
+		Set<Role> roles = new HashSet<>();
+
+		for (String siteRole : siteRoles) {
+			Role role = rolePersistence.fetchByC_N(
```
Comment (ethanbustad) [inline commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java:168]: Can't we use `roleLocalService.fetchRole` here? That'd be better

File: commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java [hunk 3]
```diff
@@ -143,26 +147,45 @@ public void addCommerceAccountUserRels(
...
+				user.getCompanyId(), siteRole);
+
+			if (role == null) {
+				continue;
+			}
+
+			if (role.getType() != RoleConstants.TYPE_SITE) {
```
Comment (ethanbustad) [inline commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java]: These 2 `continue` blocks should be combined

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by marco-leo** [comment]
@Commando18

**Comment by alessiorendina** [comment]
Fixed, thanks @ethanbustad 

cc @marco-leo

**Comment by alessiorendina** [comment]
ci:test

File: commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java (hunk 2 above)
Comment (ethanbustad) [reply commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java:168]: @Commando18 looks like this line is still the same, and SF doesn't allow it. Can you make the change?

**Comment by marco-leo** [comment]
ci:test

**Comment by ethanbustad** [comment]
@Commando18 there's still one change to make from my last review, which is causing the validation failure

**Comment by ethanbustad** [comment]
Rebased, fixed SF issues, and sent along to https://github.com/liferay/com-liferay-commerce/pull/1793.

**Pull request closed without merging**